apache/nifi
# NIFI-9174: Adding AWS SecretsManager ParamValueProvider for Stateless

#### Description of PR

_Adds a ParameterValueProvider using AWS SecretsManager_

To test:

```
aws secretsmanager create-secret --name "Context/Param" --secret-string '{ "value": "secretValue" }'
```

Create a flow with a Parameter Context named "Context" and a Parameter named "Param".  The flow could demonstrate the value by logging the parameter via `LogAttribute`.  Save the flow definition and launch stateless using the following configuration:

```
nifi.stateless.parameter.provider.AWSSecretsManager.name=AWS Secrets Manager Provider
nifi.stateless.parameter.provider.AWSSecretsManager.type=org.apache.nifi.stateless.parameter.aws.SecretsManagerParameterValueProvider
nifi.stateless.parameter.provider.AWSSecretsManager.properties.aws-configuration-file=./conf/bootstrap-aws.conf
```

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
     in the commit message?

- [ ] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.
